apps/04-manipulacion-de-listas/src/manipulating.erl

```diff
@@ -8,29 +8,77 @@
 %% @spec filter(List :: [integer()], N :: integer()) -> [integer()]
 %% @end
 %%--------------------------------------------------------------------
+filter(List, N) when is_integer(N) ->
+    filter(List, N, []);
 filter(_List, _N) ->
-    erlang:throw(not_implemented).
+    erlang:throw("NAN").
 
 %%--------------------------------------------------------------------
 %% @doc Reverse function.
 %% @spec reverse(List :: [integer()]) -> [integer()]
 %% @end
 %%--------------------------------------------------------------------
-reverse(_List) ->
-    erlang:throw(not_implemented).
+reverse(List) ->
+    reverse(List, []).
 
 %%--------------------------------------------------------------------
 %% @doc Concatenate function.
 %% @spec concatenate(ListOfLists :: [[integer()]]) -> [integer()]
 %% @end
 %%--------------------------------------------------------------------
-concatenate(_ListOfLists) ->
-    erlang:throw(not_implemented).
-
+concatenate(List)->
+    Result = concatenate(List, []),
+    reverse(Result).
+        
 %%--------------------------------------------------------------------
 %% @doc Flatten function.
 %% @spec flatten(DeepList :: [integer() | list()]) -> [integer()]
 %% @end
 %%--------------------------------------------------------------------
-flatten(_DeepList) ->
-    erlang:throw(not_implemented).
+
+flatten(List) ->
+    Result = flatten(List, []),
+    reverse(Result).
+
+%%%-----------------------------------------------------------------------------
+%%% INTERNAL FUNCTIONS
+%%%-----------------------------------------------------------------------------
+
+aux([], Acc) ->
+    Acc;
+aux([H | T], Acc) ->
+        aux(T, [H | Acc]).
+
+
+concatenate([], Acc) ->
+        Acc;
+concatenate([H | T], Acc) when is_list(H)->
+    concatenate(T, aux(H, Acc));
+concatenate(_ListOfLists, _) ->
+    erlang:throw("Not a list of lists").
+
+
+filter([], _N, Acc) ->
+    Acc;
+filter([H | T], N, Acc) when H > N ->
+    filter(T, N, Acc);
+filter([H | T], N, Acc) ->
+    filter(T, N, Acc ++ [H]).
```

Comment:
Why do you use `++` instead `[H | Acc]` ?

Reply:
For convenience, I have already changed it